sdk/src/main/scala/com/horizen/SidechainHistory.scala

```diff
@@ -35,11 +37,11 @@ class SidechainHistory private (val storage: SidechainHistoryStorage, params: Ne
   def height: Int = storage.height
   def bestBlockId: ModifierId = storage.bestBlockId
   def bestBlock: SidechainBlock = storage.bestBlock
-
+  def bestBlockInfo: SidechainBlockInfo = storage.bestBlockInfo
 
   // Note: if block already exists in History it will be declined inside NodeViewHolder before appending.
   override def append(block: SidechainBlock): Try[(SidechainHistory, ProgressInfo[SidechainBlock])] = Try {
-    validators.map(_.validate(block)).foreach {
+    validators.map(_.validate(block, this)).foreach {
```

Comment:
Introduce two separate traits for SemanticBlockValidator (validate method without history param) and HistoryBlockValidator to allow to first verify the semantic validity of the received block, then if the parent is present (or genesis), verify it with HistoryBlockValidator s  that will perform checks relying on existence of parent block (eg. WithdrawalEpochValidator), otherwise reject